### ISSUE TITLE
terraform not picking up provider details 

### ISSUE THREAD

**Issue by mkaja** [terraform-providers/terraform-provider-aws#3551]
Hi there,

I am trying to create a vpc in our aws prod account and in main.tf we have 

```
provider "aws" {
    region = "us-east-1"
   shared_credentials_file = "/Users/user-name/.aws/credentials"
   profile = "prod"

}
```
ERROR 

Error: Error refreshing state: 2 error(s) occurred:

* data.terraform_remote_state.infrastructure_prod: 1 error(s) occurred:

* data.terraform_remote_state.infrastructure_prod: data.terraform_remote_state.infrastructure_prod: error loading the remote state: AccessDenied: Access Denied
	status code: 403, request id: 01524824E7FC9D7B, host id: UJL0U8azl+lgWJIlhtfa+3Nf9MEPyGTo301IJvPi7zks1gy922B3qoBAEAVc9jtpXnG8X2ZcQfU=
* module.vpc.module.vpc.aws_default_route_table.default: 1 error(s) occurred:

* module.vpc.module.vpc.aws_default_route_table.default: aws_default_route_table.default: Default Route table not found

### Terraform Version
Run `terraform -v` to show the version 
terraform -v
Terraform v0.11.3
+ provider.aws v1.10.0

main.tf 

```
###
# This file declares all input and ouput variables
###

provider "aws" {
  region = "us-east-1"
  shared_credentials_file = "/Users/mkaja/.aws/credentials"
  profile = "prod"

}

terraform {
  required_version = "~> 0.11"

  # Modify the configuration accordingly
  backend "s3" {
      bucket = "bucket-name"
      encrypt = "true"
      key     = "infrastructure_states/us-east-1/vpc-name/terraform.tfstate"
      region  = "us-east-1"
    }
  }



data "terraform_remote_state" "infrastructure_prod" {
backend = "s3"

config {
bucket = "bucket-name"
key    = "infrastructure_states/us-east-1/infrastructure-prod/terraform.tfstate"
region = "us-east-1"
  profile = "prod"
}
}


variable "azs" {
  description = "A list of availability zones"
  type        = "list"
}
variable "cost_tags" {
  description = "A map of tags needed for cost tracking"
  type        = "map"
}
variable "default_ubuntu_ami" {
  description = "The default base Ubuntu AMI"
}
/*variable "infrastructure_nexus_sg" {
  description = "The security group id of the Nexus ELB."
}*/
variable "name" {
  description = "A prefix applied to all resources in the VPC, and it should match the name of the project"
}
variable "name_acronym" {
  description = "A shortened prefix applied to all resources that have strict length limit of the name"
}
variable "private_subnet_cidrs" {
  description = "A list of CIDR blocks for private subnets. Typically like <vpc_cidr_first_two_octet>.<even_number>.0/24. e.g. ['10.183.0.0/24', '10.183.2.0/24']"
  type        = "list"
}
variable "public_subnet_cidrs" {
  description = "A list of CIDR blocks for public subnets. Typically like <vpc_cidr_first_two_octet>.<odd_number>.0/24. e.g. ['10.183.1.0/24', '10.183.3.0/24']"
  type        = "list"
}
variable "region" {
  description = "The region the resources should be created in"
  default     = "us-east-1"
}
variable "vpc_cidr" {
  description = "A CIDR block for the VPC, typically with a subnet mask of /16"
}
variable "vpc_key_name" {
  description = "Name of the Key pair to be used to log into the instances"
}
```

### Outputs
```
output "bastion_sg" {
  value = "${module.vpc.bastion_security_group}"
}
output "bastion_sg_name" {
  value = "${module.vpc.bastion_sg_name}"
}
output "common_sg" {
  value = "${module.vpc.vpc_common_sg}"
}
output "common_sg_name" {
  value = "${module.vpc.vpc_common_sg_name}"
}
output "vpc_key_name" {
  value = "${var.vpc_key_name}"
}
output "vpc_name" {
  value = "${var.name}"
}
output "vpc_id" {
  value = "${module.vpc.vpc_id}"
}
output "vpc_region" {
  value = "${var.region}"
}
```

**Comment by mkaja** [comment]
any update on this?

**Comment by georgecodes** [comment]
Are you using the terraform-aws-vpc module? I just hit a very similar problem and it was down to us not having consumed [this change](https://github.com/terraform-aws-modules/terraform-aws-vpc/commit/6aad37fe2fc06fbb7dbd1c29432665d50cdfb836#diff-7a370d8342e7203b805911c92454f0f4).

**Comment by bflad** [comment]
The adjustment of the resource to propose resource recreation in this scenario instead of an error has been merged and will release with version 2.40.0 of the Terraform AWS Provider this week.
